derive_state_machine_future/src/codegen.rs

```diff
@@ -158,18 +157,19 @@ impl ToTokens for StateMachine<phases::ReadyForCodegen> {
                 #( #states )*
 
                 #derive
-                enum #states_enum #generics #wheres {
+                enum #states_enum #impl_generics #where_clause {
                     #( #states_variants ),*
                 }
 
                 #( #state_machine_attrs )*
                 #derive
                 #[must_use = "futures do nothing unless polled"]
-                pub struct #state_machine_ident #generics(
-                    Option<#states_enum #generics>
-                ) #wheres;
+                pub struct #state_machine_ident #impl_generics(
+                    Option<#states_enum #ty_generics>
```

Comment:
Here, the `states_enum` type is using `ty_generics`, but at its definition above, it is using `impl_generics`. Is this intentional or an oversight?

Reply:
Maybe the naming is a little bit weird (I took it from the syn documentation). The `impl_generics` are the generics with bounds, so for example `<T: Debug, E: Error>` and `ty_generics` are the same generics without the bounds. So, this is intentional.

Reply:
ah, cool, thanks!